rest/rest.go

```diff
@@ -53,7 +54,18 @@ type EndpointType client.EndpointType
 type Server struct {
 	CoreAPI     bool
 	Protocol    string
-	Address     types.AddrPort
+	Address     *types.AddrPort
```

Comment:
Does this need to be a pointer? I believe it would just be instantiated as an empty struct, which you can verify with `s.Address == types.AddrPort{}` in the validator.

Reply:
aah yes, pointer semantics doesn't make a whole lot of sense here, I will update it.